File: insertionSort.go
package main

import "fmt"

func insertionSort(arr []int) []int {

	for i := 0; i < len(arr); i++ {
		j := i - 1

		for j >= 0 && arr[j] > arr[j+1] {
			temp := arr[j+1]
			arr[j+1] = arr[j]
			arr[j] = temp

			j--
		}
	}

	return arr
}

func main() {

	input := []int{2, 3, 4, 1, 6}

	fmt.Println(insertionSort(input))
}

File: insertionSort.ts
function insertionSort(arr: number[]) {
  for (let i = 1; i < arr.length; i++) {
    let j = i - 1;

    while (j >= 0 && arr[j] > arr[j + 1]) {
      const temp = arr[j + 1];
      arr[j + 1] = arr[j];
      arr[j] = temp;
      j--;
    }
  }

  return arr;
}

console.log(insertionSort([2, 3, 4, 1, 6]));
